### DISCUSSION TITLE
Import existing folder structure into zim

### DISCUSSION THREAD

**Discussion by biosaat** [zim-desktop-wiki/zim-desktop-wiki#2030] [Q&A] [answered]
Hi, I want to import an existing structure of directories and files (of different types) into zim. 
I copied the folder structure into a subdirectory of my zim notebook. 

Obviously zim does not recognize my newly added folders as subfolders because no equally named wiki page has been created so far. 

Does there exist a script to convert my "legacy" folder structure to a zim folder structure?

I think this would at least imply
a) renaming the newly added folders according to the guidelines in [https://zim-wiki.org/manual/Help/Importing_external_files.html](https://zim-wiki.org/manual/Help/Importing_external_files.html)
b) creating a dummy txt file for each folder with the content _Content-Type: text/x-zim-wiki_

Cheers

**Comment by introt** [comment]
Hi! You may find some answers here: #1803

**Comment by biosaat** [comment, marked as answer]
Thank you, introt.

This is my solution to this problem as of today.
https://go.0xfc.de/5h3eh6

**Reply by introt** [reply]
Nice script! Since its Python, you could turn it into a plugin without too much trouble, and use the API (eg. MakeValidPageName) to ensure future compatibility :)

PS. I suggest using [Gists](gist.github.com/) for code sharing, given we're already on GitHub - those are also update-able with a version history.

**Reply by treborxy** [reply]
@biosaat : very cool! something I wished to have in my toolbox for ages ;-) Thanx a lot
